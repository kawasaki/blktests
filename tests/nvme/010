#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# This is a data verification test for block device backed ns.

. tests/nvme/rc

DESCRIPTION="run data verification fio job on NVMeOF block device-backed ns"
TIMED=1

requires() {
	_nvme_requires
	_have_fio && _have_loop
	_require_nvme_trtype_is_fabrics
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local nvmedev

	_nvmet_target_setup

	_nvme_connect_subsys "${nvme_trtype}" "${def_subsysnqn}"

	nvmedev=$(_find_nvme_dev "${def_subsysnqn}")
	_check_uuid "${nvmedev}"

	_run_fio_verify_io --size="${nvme_img_size}" \
		--filename="/dev/${nvmedev}n1"

	_nvme_disconnect_subsys "${def_subsysnqn}"

	_nvmet_target_cleanup

	echo "Test complete"
}
