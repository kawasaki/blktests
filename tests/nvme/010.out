Running nvme/010
disconnected 1 controller(s)
Test complete
